### DISCUSSION TITLE
Testing without web server

### DISCUSSION THREAD

**Discussion by stef-pellegrino** [mirumee/ariadne#497] [General]
Is it possible to test with a simple command line instead of always using web server ?

**Comment by patrys** [comment]
`uvicorn --reload` is a single command though?

**Comment by stef-pellegrino** [comment]
Yes thanks, but I do not have (or need) web server. I'm using GraphQL in batch mode...

**Comment by patrys** [comment]
Can you describe your setup in more detail? If you have a server (any kind of server) then couldn't you run it in your `__main__.py` and use `python -m <whatever>` to start it? Or use the old `if __name__ == "__main__":` at the bottom of your main script.

**Comment by rafalp** [comment]
I am guessing @stef-pellegrino is looking for something like `ariadne.py -q="query { user { pk username email } }"`?

**Comment by beneshed** [comment]
@stef-pellegrino  you could do something like the [following ](https://graphql-core-next.readthedocs.io/en/latest/usage/queries.html)

```python
import asyncio
from ariadne import make_executable_schema
from graphql import graphql

schema = make_executable_schema(FILL THIS OUT)
from graphql import graphql

async def query_artoo():
    result = await graphql(schema, """
    {
      droid(id: "2001") {
        name
        primaryFunction
      }
    }
    """)
    print(result)

asyncio.get_event_loop().run_until_complete(query_artoo())
```

But this doesn't take care of context ^^
